tests/integration/test_migrator.py

```diff
@@ -142,7 +164,41 @@ def test_continuous_migration_handles_multiple_records(app, record_1502655_and_1
     assert expected == result
 
 
+def test_continuous_migration_avoids_jamming(app, many_times_same_record):
+    with app.app_context():
+        with mock.patch.dict(app.config, {'CELERY_ALWAYS_EAGER': False}):
```

Comment:
The other thing is that you should be patching `current_app.config`, not `app.config`: I have a feeling that this was causing your woes with having to do internal imports (because you were modifying a fixture shared by other tests, not the application context on the stack).